### DISCUSSION TITLE
Packets not received by server after around 130 packets in a connection

### DISCUSSION THREAD

**Discussion by hari-19** [aiortc/aioquic#396] [Q&A]
I am running an experiment where I like to send many packets to a QUIC server. I tried opening a connection and sending packets from a client continuously (with sleep b/w packets). I noticed that the server stopped receiving the packets after around 130 packets every time. Then after a while changes, the connection state to terminated. 

The sleep times I tried: 0 sec (no sleep), 1 sec, 20 sec.

Client App code: https://gist.github.com/hari-19/5e4fdeea32b80ee25724aea38e096890
Server App code: https://gist.github.com/hari-19/d495d7760b228a6bd339e119af471d54

I have attached the relevant terminal log files below. Any help on the cause of this issue would be helpful for my research.

[server_20sec.log](https://github.com/aiortc/aioquic/files/12327425/server_20sec.log): https://gist.github.com/hari-19/e4dae248650d928760ba9c986e7ba12f
[client_20sec.log](https://github.com/aiortc/aioquic/files/12327427/client_20sec.log): https://gist.github.com/hari-19/94a4519bc1b2a321cd2149cd55277d30

[server_1sec.log](https://github.com/aiortc/aioquic/files/12327429/server_1sec.log): https://gist.github.com/hari-19/00688041c9dfe0b6372aac9cc65ccae6
[client_1sec.log](https://github.com/aiortc/aioquic/files/12327426/client_1sec.log): https://gist.github.com/hari-19/825974480ebaa0883b5a5f6ce1bc0ba7

[server_0sec.log](https://github.com/aiortc/aioquic/files/12327428/server_0sec.log): https://gist.github.com/hari-19/bb41c3b530ceb1cd06f14784e40681c0

Kudos for a great library for QUIC App development.

My setup:

The experiment was run on a Laptop with Intel i7-1260P, 16GB RAM laptop.
Client in a VMware Ubuntu 22: 8 Cores, 6 GB RAM
Server: In the Host OS (Windows 11 Home): 16 Cores, 16 GB RAM

I carried the same experiment with the server running in a laptop with Intel i5-8250U, 8GB RAM, Ubuntu 22 and Client in a laptop with Intel i7-1260P, 16GB RAM, Windows 11 Home. The observation was consistent.

**Comment by whynotvlad** [comment]
Hi, were you able to move forward with the problem? I am facing a similar issue

**Reply by hari-19** [reply]
> Hi, were you able to move forward with the problem? I am facing a similar issue

@whynotvlad 
Hi, 
My experiments didn't rely on congestion control; hence, I turned off congestion control in aioquic and proceeded. 

Check this issue: https://github.com/aiortc/aioquic/issues/395

I used time.sleep(), instead @jlaine suggested the use of aioquic.sleep(). Do check if that resolves the issue.
